### ISSUE TITLE
DeviceManager.connected_devices fails parsing USB data

### ISSUE THREAD

**Issue by kdubb** [fastlane/fastlane#7723]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

DeviceManager.connected_devices fails when parsing the USB device information.  It's expecting keys in the hash that are not there.

### Captured Output

Command Used: `build_and_test`
<details><summary>Output/Log</summary>

```

[11:20:05]: Driving the lane 'build_and_test' 🚀
[11:20:05]: ----------------------------------
[11:20:05]: --- Step: ensure_xcode_version ---
[11:20:05]: ----------------------------------
[11:20:05]: $ xcversion selected
[11:20:05]: ▸ Xcode 8.2.1
[11:20:05]: ▸ Build version 8C1002
[11:20:05]: Selected Xcode version is correct: 8.2.1
[11:20:05]: -------------------
[11:20:05]: --- Step: is_ci ---
[11:20:05]: -------------------
+---------------+----------------+
|          Lane Context          |
+---------------+----------------+
| PLATFORM_NAME |                |
| LANE_NAME     | build_and_test |
+---------------+----------------+
[11:20:06]: undefined method `each' for nil:NilClass

+------+----------------------+-------------+
|             fastlane summary              |
+------+----------------------+-------------+
| Step | Action               | Time (in s) |
+------+----------------------+-------------+
| 1    | ensure_xcode_version | 0           |
| 2    | is_ci                | 0           |
+------+----------------------+-------------+

[11:20:06]: fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...

➡️  [!] Undefined method `description' for nil:NilClass (NoMethodError)
   https://github.com/fastlane/fastlane/issues/7686 [closed] 4 💬
   5 days ago

➡️  deliver fails with: [!] undefined method `description' for nil:NilClass  
   https://github.com/fastlane/fastlane/issues/7356 [closed] 5 💬
   4 weeks ago

➡️  undefined method `<<' for nil:NilClass
   https://github.com/fastlane/fastlane/issues/6731 [closed] 7 💬
   26 Oct 2016

and 68 more at: https://github.com/fastlane/fastlane/search?q=undefined%20method%20%60each'%20for%20nil&type=Issues&utf8=✓

```

</details>

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                 | Value                                                                            |
| ------------------- | -------------------------------------------------------------------------------- |
| OS                  | 10.12.2                                                                          |
| Ruby                | 2.4.0                                                                            |
| Bundler?            | true                                                                             |
| Git                 | git version 2.10.1 (Apple Git-78)                                                |
| Installation Source | ~/BuildHome/xml-data/build-dir/MK-BV-JOB1/.vendor/bundle/ruby/2.4.0/bin/fastlane |
| Host                | Mac OS X 10.12.2 (16C67)                                                         |
| Ruby Lib Dir        | /usr/local/Cellar/ruby/2.4.0/lib                                                 |
| OpenSSL Version     | OpenSSL 1.0.2j  26 Sep 2016                                                      |
| Is contained        | false                                                                            |
| Xcode Path          | /Applications/Xcode.app/Contents/Developer/                                      |
| Xcode Version       | 8.2.1                                                                            |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
fastlane_require 'yaml'

# Global settings

workspace = "MessagesKit.xcworkspace"
edge_compositions = {
  'test' => "fastlane/edge-test-compose.yml",
  'dev' => "fastlane/edge-dev-compose.yml"
}

ios_sim_os_versions = [
  '10.2'
]

ios_sim_device_types = [
  'iPhone 7',
  'iPhone 5'
]

macos_archs = [
  'x86_64'
]


# Build Configuraion

build = YAML.load(open('../build.yaml'))

before_all do

  ensure_xcode_version(version: "8.2.1")

  # Set environment variables for docker-compose
  container = build['server']['container']
  ENV['EDGE_CONTAINER_REG'] = container['registry']
  ENV['EDGE_CONTAINER_REPO'] = container['repository']
  ENV['EDGE_CONTAINER_TAG'] = container['tag']

  # Cocoapods
  if is_ci || !File.exists?('../Pods')
    
    cocoapods

  end

end


desc "Build and execute unit & integration tests for all platforms"
lane :build_and_test do

  ios_devices = 
    FastlaneCore::DeviceManager
      .connected_devices("iOS")
      .map { |d| d.udid }

  ## Build

  ios_sim_build = platform_build(
    platform: :ios,
    simulator: true
  )

  if !FastlaneCore::Helper.is_ci?
    ios_dev_build = platform_build(
      platform: :ios,
      simulator: false
    )
  end

  macos_build = platform_build(
    platform: :mac
  )

  ## Unit tests

  platform_unit_tests(
    platform: :ios,
    sim_os_versions: ios_sim_os_versions,
    sim_device_types: ios_sim_device_types,
    build_info: ios_sim_build
  )

  if !FastlaneCore::Helper.is_ci?
    platform_unit_tests(
      platform: :ios,
      devices: ios_devices,
      build_info: ios_dev_build
    )
  end

  platform_unit_tests(
    platform: :mac,
    archs: macos_archs,
    build_info: macos_build
  )

  # Integration testing

  begin

    start_integration_environment

    platform_integration_tests(
      platform: :ios,
      sim_os_versions: ios_sim_os_versions,
      sim_device_types: ios_sim_device_types,
      build_info: ios_sim_build
    )

    if !FastlaneCore::Helper.is_ci?
      platform_integration_tests(
        platform: :ios,
        devices: ios_devices,
        build_info: ios_dev_build
      )
    end

    platform_integration_tests(
      platform: :mac,
      archs: macos_archs,
      build_info: macos_build
    )

  ensure

    stop_integration_environment

  end

end


# MARK: iOS lanes

platform :ios do

  desc "Build and execute unit & integration tests for iOS"
  lane :build_and_test do |options|

    ios_devices = 
      FastlaneCore::DeviceManager
        .connected_devices("iOS")
        .map { |d| d.udid }

    ## Build

    sim_build = platform_build(
      simulator: true
    )

    dev_build = platform_build(
      simulator: false
    )

    ## Unit tests

    platform_unit_tests(
      sim_os_versions: ios_sim_os_versions,
      sim_device_types: ios_sim_device_types,
      build_info: sim_build
    )

    platform_unit_tests(
      devices: ios_devices,
      build_info: dev_build
    )

    ## Integration tests

    begin

      start_integration_environment

      platform_integration_tests(
        sim_os_versions: ios_sim_os_versions,
        sim_device_types: ios_sim_device_types,
        build_info: sim_build
      )

      platform_integration_tests(
        devices: ios_devices,
        build_info: dev_build
      )

    ensure

      stop_integration_environment

    end

  end

end


# MARK: macOS lanes

platform :mac do

  desc "Build and execute unit & integration tests for macOS"
  lane :build_and_test do |options|

    ## Build

    platform_build

    ## Unit testing

    platform_unit_tests(
      archs: macos_archs
    )

    ## Integration testing
    
    begin
  
      start_integration_environment
      
      platform_integration_tests(
        archs: macos_archs
      )

    ensure
      
      stop_integration_environment

    end
    
  end

end


# MARK: platform lanes

private_lane :platform_build do |options|

  case options[:platform] || lane_context[:PLATFORM_NAME]
  when :ios

    build_info = xctest_build(
      workspace: workspace,
      scheme: "MessagesKit-iOS-Testing",
      simulator: options[:simulator]
    )

  when :mac

    build_info = xctest_build(
      workspace: workspace,
      scheme: "MessagesKit-macOS-Testing"
    )

  else
    raise "Invalid platform"
  end

  build_info
end

private_lane :platform_unit_tests do |options|

  run_options = options[:build_info] || {}

  run_options[:output_type] = 'Unit'
  run_options[:fail_build] = false  

  case options[:platform] || lane_context[:PLATFORM_NAME]
  when :ios

    run_options[:sim_os_versions] = options[:sim_os_versions] unless options[:sim_os_versions].nil?
    run_options[:sim_device_types] = options[:sim_device_types] unless options[:sim_device_types].nil?
    run_options[:devices] = options[:devices] unless options[:devices].nil?

    run_options[:skip_testing] = [
      'MessagesKit-iOSTests/RoundTripTests',
      'MessagesKit-iOSTests/MessagesKitTests'
    ]

  when :mac

    run_options[:archs] = options[:archs]

    run_options[:skip_testing] = [
      'MessagesKit-macOSTests/RoundTripTests',
      'MessagesKit-macOSTests/MessagesKitTests'
    ]

  else
    raise "Invalid platform"
  end
    
  xctest_run(**run_options)

end


private_lane :platform_integration_tests do |options|

  run_options = options[:build_info] || {}

  run_options[:output_type] = 'Integration'
  run_options[:fail_build] = false  
  
  server_info = options[:server_info] || lane_context[:server_info]
  run_options[:launch_args] = [
    "-io.retxt.test.edge.host.name #{server_info[:edge][:host]}",
    "-io.retxt.test.edge.host.port #{server_info[:edge][:port]}",
    "-io.retxt.test.apns.simulator.host.name #{server_info[:apns][:host]}",
    "-io.retxt.test.apns.simulator.host.port #{server_info[:apns][:port]}"
  ]

  case options[:platform] || lane_context[:PLATFORM_NAME]
  when :ios
    
    run_options[:sim_os_versions] = options[:sim_os_versions] unless options[:sim_os_versions].nil?
    run_options[:sim_device_types] = options[:sim_device_types] unless options[:sim_device_types].nil?
    run_options[:devices] = options[:devices] unless options[:devices].nil?

    run_options[:only_testing] = [
      'MessagesKit-iOSTests/RoundTripTests',
      'MessagesKit-iOSTests/MessagesKitTests'
    ]

  when :mac

    run_options[:archs] = options[:archs]

    run_options[:only_testing] = [
      'MessagesKit-macOSTests/RoundTripTests',
      'MessagesKit-macOSTests/MessagesKitTests'
    ]

  else
    raise "Invalid platform"
  end

  xctest_run(**run_options)

end

# MARK: Server environment lanes

private_lane :start_integration_environment do |options|

  integration_id = (0...8).map { (65 + rand(26)).chr }.join.downcase

  server_info = start_environment(
    type: :test,
    name: integration_id,
    pull: true
  )

  lane_context[:integration_id] = integration_id
  lane_context[:server_info] = server_info

end

private_lane :stop_integration_environment do |options|

  stop_environment(
    type: :test,
    name: lane_context[:integration_id]
  )

end

desc "Starts an Edge server environment (Edge & DB containers)"
lane :start_environment do |options|

  type = options[:type] || :dev
  name = options[:name] || type
  pull = options[:pull].nil? ? true : options[:pull]

  files = [
    edge_compositions[type.to_s]
  ]

  # Pull image (if requested)
  if pull
    docker_compose(
      files: files,
      command: "pull"
    )
  end

  docker_compose(
    files: files,
    command: "up",
    project: name.to_s
  )
  
  edge_port_info = docker_port(
    container: "#{name}_edge_1",
    port: 8080,
  )

  apns_port_info = docker_port(
    container: "#{name}_apns_1",
    port: 8090,
  )

  # Ensure we have a publicly usable address
  if edge_port_info[:host] == '0.0.0.0'
    ip = Socket.ip_address_list.detect{|intf| intf.ipv4_private?}
    edge_port_info[:host] = ip.ip_address
  end
  if apns_port_info[:host] == '0.0.0.0'
    ip = Socket.ip_address_list.detect{|intf| intf.ipv4_private?}
    apns_port_info[:host] = ip.ip_address
  end

  http_service_available(
    target: "http://#{edge_port_info[:host]}:#{edge_port_info[:port]}/admin/healthcheck",
    timeout: 40
  )
  
  # Return resolved ports
  {
    edge: edge_port_info,
    apns: apns_port_info
  }
end


desc "Stops an Edge server environment"
lane :stop_environment do |options|

  type = options[:type] || :dev
  name = options[:name] || type

  files = [
    edge_compositions[type.to_s]
  ]

  docker_compose(
    files: files,
    command: "down",
    project: name.to_s
  )
end
```
</details>

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status       |
| -------- | ------- | ------------------- |
| fastlane | 2.3.1   | 🚫 Update availaible |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.1.0        |
| bundler                   | 1.13.7       |
| io-console                | 0.4.6        |
| openssl                   | 2.0.2        |
| CFPropertyList            | 2.3.4        |
| i18n                      | 0.7.0        |
| minitest                  | 4.7.5        |
| multi_json                | 1.12.1       |
| thread_safe               | 0.3.5        |
| tzinfo                    | 0.3.52       |
| activesupport             | 4.0.13       |
| public_suffix             | 2.0.4        |
| addressable               | 2.5.0        |
| babosa                    | 1.0.2        |
| claide                    | 1.0.1        |
| fuzzy_match               | 2.0.4        |
| nap                       | 1.1.0        |
| cocoapods-core            | 1.1.1        |
| cocoapods-deintegrate     | 1.0.1        |
| cocoapods-downloader      | 1.1.3        |
| cocoapods-plugins         | 1.0.0        |
| cocoapods-search          | 1.0.0        |
| cocoapods-stats           | 1.0.0        |
| netrc                     | 0.7.8        |
| cocoapods-trunk           | 1.1.2        |
| cocoapods-try             | 1.1.0        |
| colored                   | 1.2          |
| escape                    | 0.0.4        |
| fourflusher               | 2.0.1        |
| gh_inspector              | 1.0.2        |
| molinillo                 | 0.5.4        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.4.2        |
| cocoapods                 | 1.1.1        |
| highline                  | 1.7.8        |
| commander                 | 4.4.3        |
| unf_ext                   | 0.0.7.2      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20161129 |
| dotenv                    | 2.1.1        |
| excon                     | 0.54.0       |
| multipart-post            | 2.0.0        |
| faraday                   | 0.10.0       |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.10.1       |
| fastimage                 | 2.0.1        |
| jwt                       | 1.5.6        |
| little-plugger            | 1.1.4        |
| logging                   | 2.1.0        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| hurley                    | 0.2          |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| uber                      | 0.0.15       |
| representable             | 2.3.0        |
| retriable                 | 2.1.0        |
| google-api-client         | 0.9.20       |
| json                      | 2.0.2        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| plist                     | 3.2.0        |
| rubyzip                   | 1.2.0        |
| security                  | 0.1.3        |
| slack-notifier            | 1.5.1        |
| terminal-notifier         | 1.7.1        |
| unicode-display_width     | 1.1.2        |
| terminal-table            | 1.7.3        |
| word_wrap                 | 1.0.0        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.4        |
| xcpretty-travis-formatter | 0.0.4        |
| xcode-install             | 2.1.0        |
</details>

*generated on:* **2017-01-03**
</details>

**Comment by kdubb** [comment]
<details><summary>USB output that fails parsing</summary>

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<array>
	<dict>
		<key>_SPCommandLineArguments</key>
		<array>
			<string>/usr/sbin/system_profiler</string>
			<string>-nospawn</string>
			<string>-xml</string>
			<string>SPUSBDataType</string>
			<string>-detailLevel</string>
			<string>full</string>
		</array>
		<key>_SPCompletionInterval</key>
		<real>0.058021008968353271</real>
		<key>_SPResponseTime</key>
		<real>0.14318501949310303</real>
		<key>_dataType</key>
		<string>SPUSBDataType</string>
		<key>_detailLevel</key>
		<integer>-1</integer>
		<key>_items</key>
		<array>
			<dict>
				<key>_items</key>
				<array>
					<dict>
						<key>Built-in_Device</key>
						<string>Yes</string>
						<key>_name</key>
						<string>hub_device</string>
						<key>bcd_device</key>
						<string>0.00</string>
						<key>bus_power</key>
						<string>500</string>
						<key>bus_power_used</key>
						<string>0</string>
						<key>device_speed</key>
						<string>high_speed</string>
						<key>extra_current_used</key>
						<string>0</string>
						<key>location_id</key>
						<string>0x1a100000 / 1</string>
						<key>product_id</key>
						<string>0x0024</string>
						<key>vendor_id</key>
						<string>0x8087  (Intel Corporation)</string>
					</dict>
				</array>
				<key>_name</key>
				<string>USB20Bus</string>
				<key>host_controller</key>
				<string>AppleUSBEHCIPCI</string>
				<key>pci_device</key>
				<string>0x1e2d </string>
				<key>pci_revision</key>
				<string>0x0004 </string>
				<key>pci_vendor</key>
				<string>0x8086 </string>
			</dict>
			<dict>
				<key>_items</key>
				<array>
					<dict>
						<key>Built-in_Device</key>
						<string>Yes</string>
						<key>_items</key>
						<array>
							<dict>
								<key>Built-in_Device</key>
								<string>Yes</string>
								<key>_items</key>
								<array>
									<dict>
										<key>Built-in_Device</key>
										<string>Yes</string>
										<key>_name</key>
										<string>IR Receiver</string>
										<key>bcd_device</key>
										<string>1.00</string>
										<key>bus_power</key>
										<string>500</string>
										<key>bus_power_used</key>
										<string>100</string>
										<key>device_speed</key>
										<string>low_speed</string>
										<key>extra_current_used</key>
										<string>0</string>
										<key>location_id</key>
										<string>0x1d182000 / 4</string>
										<key>manufacturer</key>
										<string>Apple, Inc.</string>
										<key>product_id</key>
										<string>0x8242</string>
										<key>vendor_id</key>
										<string>apple_vendor_id</string>
									</dict>
									<dict>
										<key>Built-in_Device</key>
										<string>Yes</string>
										<key>_items</key>
										<array>
											<dict>
												<key>Built-in_Device</key>
												<string>Yes</string>
												<key>_name</key>
												<string>Bluetooth USB Host Controller</string>
												<key>bcd_device</key>
												<string>1.31</string>
												<key>bus_power</key>
												<string>500</string>
												<key>bus_power_used</key>
												<string>0</string>
												<key>device_speed</key>
												<string>full_speed</string>
												<key>extra_current_used</key>
												<string>0</string>
												<key>location_id</key>
												<string>0x1d181300 / 6</string>
												<key>manufacturer</key>
												<string>Apple Inc.</string>
												<key>product_id</key>
												<string>0x828a</string>
												<key>vendor_id</key>
												<string>apple_vendor_id</string>
											</dict>
										</array>
										<key>_name</key>
										<string>BRCM20702 Hub</string>
										<key>bcd_device</key>
										<string>1.00</string>
										<key>bus_power</key>
										<string>500</string>
										<key>bus_power_used</key>
										<string>94</string>
										<key>device_speed</key>
										<string>full_speed</string>
										<key>extra_current_used</key>
										<string>0</string>
										<key>location_id</key>
										<string>0x1d181000 / 3</string>
										<key>manufacturer</key>
										<string>Apple Inc.</string>
										<key>product_id</key>
										<string>0x4500</string>
										<key>vendor_id</key>
										<string>0x0a5c  (Broadcom Corp.)</string>
									</dict>
								</array>
								<key>_name</key>
								<string>hub_device</string>
								<key>bcd_device</key>
								<string>b.b3</string>
								<key>bus_power</key>
								<string>500</string>
								<key>bus_power_used</key>
								<string>2</string>
								<key>device_speed</key>
								<string>high_speed</string>
								<key>extra_current_used</key>
								<string>0</string>
								<key>location_id</key>
								<string>0x1d180000 / 2</string>
								<key>product_id</key>
								<string>0x2512</string>
								<key>vendor_id</key>
								<string>0x0424  (SMSC)</string>
							</dict>
						</array>
						<key>_name</key>
						<string>hub_device</string>
						<key>bcd_device</key>
						<string>0.00</string>
						<key>bus_power</key>
						<string>500</string>
						<key>bus_power_used</key>
						<string>0</string>
						<key>device_speed</key>
						<string>high_speed</string>
						<key>extra_current_used</key>
						<string>0</string>
						<key>location_id</key>
						<string>0x1d100000 / 1</string>
						<key>product_id</key>
						<string>0x0024</string>
						<key>vendor_id</key>
						<string>0x8087  (Intel Corporation)</string>
					</dict>
				</array>
				<key>_name</key>
				<string>USB20Bus</string>
				<key>host_controller</key>
				<string>AppleUSBEHCIPCI</string>
				<key>pci_device</key>
				<string>0x1e26 </string>
				<key>pci_revision</key>
				<string>0x0004 </string>
				<key>pci_vendor</key>
				<string>0x8086 </string>
			</dict>
			<dict>
				<key>_name</key>
				<string>USB30Bus</string>
				<key>host_controller</key>
				<string>AppleUSBXHCIPPT</string>
				<key>pci_device</key>
				<string>0x1e31 </string>
				<key>pci_revision</key>
				<string>0x0004 </string>
				<key>pci_vendor</key>
				<string>0x8086 </string>
			</dict>
		</array>
		<key>_parentDataType</key>
		<string>SPHardwareDataType</string>
		<key>_properties</key>
		<dict>
			<key>1284DeviceID</key>
			<dict>
				<key>_order</key>
				<string>13</string>
			</dict>
			<key>_name</key>
			<dict>
				<key>_isColumn</key>
				<string>YES</string>
				<key>_isOutlineColumn</key>
				<string>YES</string>
				<key>_order</key>
				<string>0</string>
			</dict>
			<key>bcd_device</key>
			<dict>
				<key>_order</key>
				<string>3</string>
				<key>_suppressLocalization</key>
				<string>YES</string>
			</dict>
			<key>bsd_name</key>
			<dict>
				<key>_order</key>
				<string>42</string>
			</dict>
			<key>bus_power</key>
			<dict>
				<key>_order</key>
				<string>8</string>
			</dict>
			<key>bus_power_desired</key>
			<dict>
				<key>_order</key>
				<string>9</string>
			</dict>
			<key>bus_power_used</key>
			<dict>
				<key>_order</key>
				<string>10</string>
			</dict>
			<key>detachable_drive</key>
			<dict>
				<key>_order</key>
				<string>39</string>
			</dict>
			<key>device_manufacturer</key>
			<dict>
				<key>_order</key>
				<string>20</string>
			</dict>
			<key>device_model</key>
			<dict>
				<key>_order</key>
				<string>22</string>
			</dict>
			<key>device_revision</key>
			<dict>
				<key>_order</key>
				<string>24</string>
			</dict>
			<key>device_serial</key>
			<dict>
				<key>_order</key>
				<string>26</string>
			</dict>
			<key>device_speed</key>
			<dict>
				<key>_order</key>
				<string>5</string>
			</dict>
			<key>disc_burning</key>
			<dict>
				<key>_order</key>
				<string>32</string>
			</dict>
			<key>extra_current_used</key>
			<dict>
				<key>_order</key>
				<string>11</string>
			</dict>
			<key>file_system</key>
			<dict>
				<key>_order</key>
				<string>40</string>
			</dict>
			<key>free_space</key>
			<dict>
				<key>_deprecated</key>
				<true/>
				<key>_order</key>
				<string>19</string>
			</dict>
			<key>free_space_in_bytes</key>
			<dict>
				<key>_isByteSize</key>
				<true/>
				<key>_order</key>
				<string>19</string>
			</dict>
			<key>location_id</key>
			<dict>
				<key>_order</key>
				<string>7</string>
			</dict>
			<key>manufacturer</key>
			<dict>
				<key>_order</key>
				<string>6</string>
			</dict>
			<key>mount_point</key>
			<dict>
				<key>_order</key>
				<string>44</string>
			</dict>
			<key>optical_drive_type</key>
			<dict>
				<key>_order</key>
				<string>30</string>
			</dict>
			<key>optical_media_type</key>
			<dict>
				<key>_order</key>
				<string>31</string>
			</dict>
			<key>product_id</key>
			<dict>
				<key>_order</key>
				<string>1</string>
			</dict>
			<key>removable_media</key>
			<dict>
				<key>_order</key>
				<string>34</string>
			</dict>
			<key>serial_num</key>
			<dict>
				<key>_order</key>
				<string>4</string>
				<key>_suppressLocalization</key>
				<string>YES</string>
			</dict>
			<key>size</key>
			<dict>
				<key>_deprecated</key>
				<true/>
				<key>_order</key>
				<string>18</string>
			</dict>
			<key>size_in_bytes</key>
			<dict>
				<key>_isByteSize</key>
				<true/>
				<key>_order</key>
				<string>18</string>
			</dict>
			<key>sleep_current</key>
			<dict>
				<key>_order</key>
				<string>12</string>
			</dict>
			<key>vendor_id</key>
			<dict>
				<key>_order</key>
				<string>2</string>
			</dict>
			<key>volumes</key>
			<dict>
				<key>_detailLevel</key>
				<string>0</string>
			</dict>
			<key>writable</key>
			<dict>
				<key>_order</key>
				<string>36</string>
			</dict>
		</dict>
		<key>_timeStamp</key>
		<date>2017-01-03T19:17:20Z</date>
		<key>_versionInfo</key>
		<dict>
			<key>com.apple.SystemProfiler.SPUSBReporter</key>
			<string>900.4.1</string>
		</dict>
	</dict>
</array>
</plist>
```

</details>

**Comment by mfurtak** [comment]
Thanks for the report, and the problematic input. That made it really easy to reproduce :100:

I'll put together a fix 👍

**Comment by kdubb** [comment]
@mfurtak Maybe check out #7675 too. It's was generated on another machine but a related bug.

**Comment by mfurtak** [comment]
Fixed in #7725 and will go out in the next release 🚀
